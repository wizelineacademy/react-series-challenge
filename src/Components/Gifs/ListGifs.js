import React from "react";
import { connect } from "react-redux";

import Trending from "../Trending/TrendingGifs";
import FilteredGifs from "../Filtered/FilteredGifs";

const ListGifs = ({ trendGifs, searchGifs, onRequestTrendGifs }) =>
  searchGifs.data && searchGifs.data.length ? (
    <FilteredGifs {...searchGifs} />
  ) : (
    <Trending {...trendGifs} onRequestTrendGifs={onRequestTrendGifs} />
  );

const mapStateToProps = state => {
  const { trendGifs, searchGifs } = state;
  return {
    trendGifs,
    searchGifs
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onRequestTrendGifs: () => dispatch({ type: "API_CALL_REQUEST" })
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ListGifs);
