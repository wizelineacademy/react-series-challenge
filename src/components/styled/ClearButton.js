import styled from 'styled-components'

const ClearButton = styled.button`
  font-family: 'Font Awesome 5 Free';
  font-size: 20px;
  padding: 8px;
  border: none;
  background-color: transparent;
  outline: none;
  cursor: pointer;

  :before {
    content: '\f00d';
  }

  &.hidden {
    display: none;
  }
`

export default ClearButton;
