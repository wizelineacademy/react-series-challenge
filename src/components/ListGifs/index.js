import ListGifs from './ListGifs';

export default ListGifs;