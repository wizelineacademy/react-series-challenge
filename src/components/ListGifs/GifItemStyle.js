import styled from "styled-components";

const ItemGif = styled.div`
    position: relative;
`

const Image = styled.img`
    height: 200px;
    width: 200px;
    padding: 0.5rem;
    display: flex;
    object-fit: cover;
    justify-content: flex-end;

`;

export { ItemGif, Image };
