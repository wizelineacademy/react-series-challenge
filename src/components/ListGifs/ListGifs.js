/**
 * List component to show favorites or trending gifs according to the request
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';

import favoritesActions from '../../actions/favorites';

import { ListGifStyled, FavButton } from './ListGifs.styled';

class ListGifs extends Component {

    checkIsFav(tgId){
        for(let i=0, len = this.props.favorites.originalList.data.length; i<len; i++){
            if(this.props.favorites.originalList.data[i].id === tgId) {
                return true;
            }
        }
        return false;
    }

    render() {
        const { dataSource } = this.props;
        let dataSourceObj = null;
        let listObj = [];

        switch (dataSource) {
            case 'favorites':
                dataSourceObj = this.props.favorites;
                break
            default:
                dataSourceObj = this.props.giphy;
        }
        if (!dataSourceObj || !dataSourceObj.list) {
            return null;
        }
        if (dataSourceObj.list && dataSourceObj.fetching) {
            return <div>Cargando...</div>
        }
        
        for(let i = 0, len=dataSourceObj.list.data.length; i<len; i++){
            const item = dataSourceObj.list.data[i];
            let isFav = false
            if (this.props.favorites.originalList.data) {
                isFav = this.checkIsFav(item.id);
            }
            listObj.push(<li key={item.id}>
                <FavButton
                    faved={isFav}
                    onClick={event => {
                        if (!isFav) {
                            this.props.addFavorite(item);
                        } else {
                            this.props.removeFavorite(item.id);
                        }
                    }}>FavMe</FavButton>
                    <Link to={'/gif/'+item.id}>
                        <img src={item.images.preview_gif.url} alt={item.title} />
                    </Link>
            </li>);
        }
        return (
            <ListGifStyled>
                {listObj}
            </ListGifStyled>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        giphy: state.giphy,
        favorites: state.favorites
    };
}

const mapDispatchToProps = (dispatch) => {
    const { addFavorite, removeFavorite } = favoritesActions.creators;

    return bindActionCreators({
        addFavorite,
        removeFavorite
    }, dispatch);
}

export {
    ListGifs
}

export default connect(mapStateToProps, mapDispatchToProps)(ListGifs);
