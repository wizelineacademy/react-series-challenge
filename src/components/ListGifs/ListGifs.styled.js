import styled from 'styled-components';
import { COLORS } from '../../utils/ConstantsGraphics';

const ListGifStyled = styled.ul`
display: grid;
grid-template-columns: repeat(5, 1fr);
list-style: none;
padding: 0;
margin: 0;
@media only screen and (max-width: 64em) {
    grid-template-columns: repeat(4, 1fr);
}
li{
    background-color: #000;
    cursor: pointer;
    position: relative;
    transition: transform .15s ease-in;
    &:hover {
        transform:scale(1.25, 1.25);
        z-index: 1;
    }
    button {
        position: absolute;
        top: .5em;
        right: .5em;
    }
    img {
        display: block;
        width: 100%;
        height: 100%;
        object-fit: cover;
    }
}
`;

const FavButton = styled.button.attrs({
    type: 'button'
}) `
    background: none;
    background-color: ${props => props.faved ? COLORS.primary : COLORS.primaryLighten};
    border: ${props => props.faved ? '#FFF solid .125em' : 0};
    border-radius: .5em;
    box-shadow: ${props => props.faved ? '0 .25em .25em rgba(0,0,0,.6)' : 'none'};
    color: #FFF;
    cursor: pointer;
    font-size: .85em;
    webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    outline: none;
    opacity: ${props => props.faved ? '1' : '.5'};
    transition: background-color .3s ease-in;
    &::-moz-focus-inner {
        border: 0;
        padding: 0;
    }
    &:hover {
        background-color: ${COLORS.primary};
    }
`;

export {
    ListGifStyled,
    FavButton
};