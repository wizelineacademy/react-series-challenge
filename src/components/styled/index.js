import ClearButton from './ClearButton';
import DetailsDiv from './DetailsDiv';
import FavButton from './FavButton';
import Header from './Header';
import ItemDiv from './ItemDiv';
import ItemListDiv from './ItemListDiv';
import SearchButton from './SearchButton';
import SearchDiv from './SearchDiv';
import SearchText from './SearchText';

export {
  ClearButton,
  DetailsDiv,
  FavButton,
  Header,
  ItemDiv,
  ItemListDiv,
  SearchButton,
  SearchDiv,
  SearchText
}
