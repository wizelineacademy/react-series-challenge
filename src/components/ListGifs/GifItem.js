import React from "react";
import FavoriteButton from "../FavoriteButton/FavoriteButton";
import { ItemGif, Image } from "./GifItemStyle";

const GifItem = ({ gif, favoriteGifs }) => 
   <ItemGif>
    <Image src={gif.images.preview_webp.url} alt="Gif item" />
    <FavoriteButton gif={gif} favorite ={favoriteGifs[gif.id]} />
  </ItemGif>;

export default GifItem;