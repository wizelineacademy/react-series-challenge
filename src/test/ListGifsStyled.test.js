import React from 'react';
import { shallow } from 'enzyme';
import { FavButton as PureComp } from '../components/ListGifs/ListGifs.styled';

const baseProps = {
    faved: true
}

describe('test ListGifs Render', () => {
    it('render structure', () => {
      const component = shallow(<PureComp  {...baseProps} />);
      expect(component).toMatchSnapshot();
    });
  });