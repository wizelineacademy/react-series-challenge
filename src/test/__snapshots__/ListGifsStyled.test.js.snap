// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ListGifs Render render structure 1`] = `
<button
  className="sc-bwzfXH fXjREY"
  type="button"
/>
`;
